### DISCUSSION TITLE
Expose BTCPay Server and receive Lightning payments

### DISCUSSION THREAD

**Discussion by danielcharrua** [Radiokot/umbrel-cloudflared#9] [Q&A]
Hello, first of all, great app!

I was doing all this manually some year ago with BTCPay Server and a Cloudflared tunnel but found that it doesn't work with lightning payments.

You can follow my issue here https://github.com/btcpayserver/btcpayserver-docker/issues/716

Just wanted to be sure if your app and development have solved this or is still a tunneling issue.

Thank you.

**Comment by Radiokot** [comment]
Hi Daniel, thank you for checking out the app.
I've never had my own LN node, but reading through your issue I see that you are trying to connect lightning-cli to **Cloudflare IP address** with a custom port. This will never work:
1. The service behind Cloudflare can only be accessed by a hostname;
2. Cloudflare tunnel only serves HTTP(S) traffic. I doubt Lightning uses HTTP protocol.

I use Cloudflare tunnel only to expose HTTP services. For non-HTTP connections, like Bitcoin P2P or Electrum server, I run a wireguard tunnel to a VPS with a white IP which can accept any connection.
